atspi-connection/src/p2p.rs

```diff
@@ -0,0 +1,768 @@
+//! Extends the `AccessibilityConnection` with P2P capabilities.
+//!
+//! # Considerations on using executors and P2P
+//!
+//! Every connection has a zbus `Executor` instance, on which tasks can be launched. Internally, zbus uses this executor for all sorts of tasks: listening for D-Bus signals,
+//! keeping property caches up to date, handling timeout errors, etc.
+//!
+//! When zbus users use `tokio` (zbus feature "tokio" set), zbus will latch onto the tokio runtime.
+//! The `Executor` instance will be empty and all zbus tasks are run on the user's tokio runtime.
+//! However, when using any other executor (smol, glommio, async-std, etc.), each `Connection` will spin up a thread with an `async_executor::Executor`.
+//!
+//! Typically an application will have a single connection, but with P2P, your application will have a connection with each application that supports it.
+//! Consequently, on anything but tokio, applications will get an extra thread with an `async_executor` for each connection!
+//! (So picking smol won't necessarily make your application small in the context of P2P.)
+
+use async_lock::Mutex;
```

Comment:
You do not need [and are encouraged not to use (from tokio)](https://docs.rs/tokio/latest/tokio/sync/struct.Mutex.html#which-kind-of-mutex-should-you-use) async locks, where possible. There are also some functions like `remove_unique`, `peers` and a few others that do not need to me async (or are only async due to the async Mutex). Only use async when it is a hard requirement.

Anyways, you don't even hold a value across an await paint so you should be fine :)

Reply:
You are right, they are not needed.
I am sure I did not put them there out of precaution. I think I refactored away it's necessity.
That's great because it simplifies code, removes async requirements on a handful of methods and drops the dependency on async-lock.